### PR TITLE
chore: add replica-priority flag

### LINKED ISSUE #3202: slave-priority / replica priority like redis configuration
**Did you search GitHub Issues and GitHub Discussions First?**
Yes, I have searched for the issue and have had discussions using Discord and I was recommended to open the issue here

**Is your feature request related to a problem? Please describe.**
yes, that's right, I use dragonfly replication / dragonfly cluster with the master slave method which can do auto failover with sentinel merging to find out the master and slave on dragonfly, so I have 3 nodes that are used as replication with 1 master, then can do auto failover if The master has a problem, the funds will make the slave the next master, with this configuration I have problems because I can't determine which one will be the next master from the best slave, because the failover selects the slave as the next master randomly.

The configuration that can determine which slave will be made the next master is the slave-priority / replica-priority configuration which I didn't find in Dragonfly, but I can find this configuration in Redis.

If this configuration can be applied to Dragonfly, the slave-priority / replica-priority configuration is very helpful in implementing cluster or master slave replication with auto failover, and can determine the next slave that you want to make master.

**Describe the solution you'd like**
I really hope that the slave-priority or replica-priority configuration is available on Dragonfly, making it easier to do failover or of course switchover so that I can find out the next master of the selected slave

**Describe alternatives you've considered**
Configuration slave-priority / replica-priority like redis

**Additional context**
gambar

Thanks you

Regards
Levi khulaifi

### PR DESCRIPTION
**Pull request by kostasrim**
resolves #3202

Allows sentinel to do automatic failover based on the score of each replica on multi replica setup

* add replica-priority flag
* print it if node is replica as slave_priority

### CODE REVIEW

**Comment by romange** [comment]
Thanks for the quick turnaround. We have sentinel tests in pytests Care to add this use case there?

**Comment by kostasrim** [comment]
> Thanks for the quick turnaround. We have sentinel tests in pytests Care to add this use case there?

ofc!

**Comment by levikhulaifi** [comment]
@kostasrim Thank you very much

File: tests/dragonfly/sentinel_test.py [hunk 1]
```diff
@@ -239,3 +241,54 @@ async def test_master_failure(df_local_factory, sentinel, port_picker):
...
+        lambda: sentinel.master(),
+        lambda m: m["num-slaves"] == "3",
+        timeout_sec=15,
+        timeout_msg="Timeout waiting for sentinel to pick up replica.",
+    )
+
+    # Simulate master failure.
+    master.stop()
+
+    # Verify replica pormoted.
```
Comment (romange) [inline tests/dragonfly/sentinel_test.py:288]: promoted

**Review by romange — APPROVED** [review]

File: tests/dragonfly/sentinel_test.py (hunk 1 above)
Comment (kostasrim) [reply tests/dragonfly/sentinel_test.py:288]: one sec will push the wording

**Review by romange — APPROVED** [review]

**Pull request merged**